crates/holochain/src/sweettest/sweet_conductor.rs

```diff
@@ -157,12 +180,17 @@ impl SweetConductor {
         };
 
         tracing::info!(?config);
-        let dir = TestDir::new(test_db_dir());
-        assert!(
-            dir.read_dir().unwrap().next().is_none(),
-            "Test dir not empty - {:?}",
-            dir.to_path_buf()
-        );
+
+        if !with_metrics {
+            // this check doesn't work with metrics enabled,
+            // because the metrics database is in there...
```

Comment:
This check is to make sure sweettest doesn't re-use an existing directory - why have metrics already initialized at this point? Nothing should have initialized until sweettest makes the decision to start Holochain

Reply:
yaa... i don't think that call serves any purpose, it's right after the test_db request which makes a new temp dir, so it should always be empty, right?

In any case, I moved the check up above the metrics initialization so it's no longer an issue.

Reply:
I added it because we aren't letting the system allocate temporary directories, we're creating them directly using nanoid. I wasn't sure how that behaved across multiple test runs on the same machine. Given we aren't hitting this error I'd be fine with the code being removed now